Hazel/src/Platform/Windows/WindowsWindow.cpp

```diff
@@ -48,28 +48,84 @@ namespace Hazel {
 			s_GLFWInitialized = true;
 		}
 
-		m_Window = glfwCreateWindow((int)props.Width, (int)props.Height, m_Data.Title.c_str(), nullptr, nullptr);
+		GLFWmonitor* windowRenderScreen = glfwGetPrimaryMonitor(); // where the window will be rendered
+		const GLFWvidmode* baseVideoMode = glfwGetVideoMode(windowRenderScreen);
+
+		// Set the actual width and height
+		switch (m_Data.Mode)
+		{
+			case WindowMode::Borderless:
+			case WindowMode::Fullscreen:
+				m_Data.Width = baseVideoMode->width;
+				m_Data.Height = baseVideoMode->height;
+				break;
+			case WindowMode::Windowed:
+				m_Data.Width = props.Width;
+				m_Data.Height = props.Height;
+				break;
+		}
+
+		HZ_CORE_INFO("Creating window '{0}' ({1}, {2})", m_Data.Title, m_Data.Width, m_Data.Height);
 
+		glfwWindowHint(GLFW_DECORATED, m_Data.Mode == WindowMode::Windowed ? GLFW_TRUE : GLFW_FALSE);
+		glfwWindowHint(GLFW_RESIZABLE, m_Data.Mode == WindowMode::Windowed ? GLFW_TRUE : GLFW_FALSE);
+		glfwWindowHint(GLFW_FLOATING, m_Data.Mode == WindowMode::Fullscreen ? GLFW_TRUE : GLFW_FALSE);
+		glfwWindowHint(GLFW_AUTO_ICONIFY, m_Data.Mode == WindowMode::Fullscreen ? GLFW_TRUE : GLFW_FALSE);
+		m_Window = glfwCreateWindow((int)m_Data.Width, (int)m_Data.Height, m_Data.Title.c_str(),
+			m_Data.Mode == WindowMode::Windowed ? nullptr : windowRenderScreen, nullptr);
```

Comment:
No because if you set it to windowRenderScreen it will become full-screened. When set to ``nullptr`` it is windowed. It sets to ``nullptr`` when it's meant to be a window.

Reply:
Readability.

Reply:
Both seem equally readable to me.

Reply:
Instead of saying "if we are in windowed mode we set it to nullptr" am I saying "if we are not in windowed mode, we have to provide a screen"... Both are equaly readable yes, but somehow it shows some logic behind it... Doesn't have any code impact tho...

Reply:
I think this change is useless, both are fine and both make sense.